packages/driver/src/cy/multidomain/index.ts

```diff
@@ -91,54 +123,52 @@ export function addCommands (Commands, Cypress: Cypress.Cypress, cy: Cypress.cy)
         }
       }
 
-      // @ts-ignore
-      Cypress.on('cross:domain:command:enqueued', addCommand)
-      // @ts-ignore
-      Cypress.on('cross:domain:command:update', updateCommand)
+      communicator.on('command:enqueued', addCommand)
+      communicator.on('command:update', updateCommand)
 
-      return new Bluebird((resolve) => {
-        // @ts-ignore
-        Cypress.once('cross:domain:ran:domain:fn', () => {
-          resolve()
-        })
+      return new Bluebird((resolve, reject) => {
+        communicator.once('ran:domain:fn', resolve)
 
-        // @ts-ignore
-        Cypress.once('cross:domain:queue:finished', () => {
-          // @ts-ignore
-          Cypress.off('cross:domain:command:enqueued', addCommand)
-          // @ts-ignore
-          Cypress.off('cross:domain:command:update', updateCommand)
+        communicator.once('queue:finished', () => {
+          communicator.off('command:enqueued', addCommand)
+          communicator.off('command:update', updateCommand)
         })
 
         // fired once the spec bridge is set up and ready to
         // receive messages
-        // @ts-ignore
-        Cypress.once('cross:domain:bridge:ready', () => {
+        communicator.once('bridge:ready', () => {
+          state('readyForMultidomain', true)
           // let the proxy know to let the response for the secondary
           // domain html through, so the page will finish loading
-          // @ts-ignore
-          Cypress.backend('ready:for:domain')
+          Cypress.backend('ready:for:domain', { shouldInject: true })
         })
 
-        // @ts-ignore
-        cy.once('cross:domain:window:load', () => {
+        cy.once('internal:window:load', ({ type }) => {
+          if (type !== 'cross:domain') return
+
           // once the secondary domain page loads, send along the
           // user-specified callback to run in that domain
-          Cypress.action('cy:cross:domain:message', {
-            message: 'run:domain:fn',
-            // the log count needs to be synced between domains so logs
-            // are guaranteed to have unique ids
-            logCounter: $Log.getCounter(),
-            fn: fn.toString(),
-          })
+          try {
+            communicator.toSpecBridge('run:domain:fn', {
+              data,
+              fn: callbackFn.toString(),
+            })
+          } catch (err: any) {
+            const wrappedErr = $errUtils.errByPath('switchToDomain.run_domain_fn_errored', {
+              error: err.message,
+            })
+
+            reject(wrappedErr)
```

Comment:
this doesn't need to `return reject(wrappedErr)`, right?

Reply:
Right, we still want to reset the multi-domain-related flags below. I think I'll move those into a `finally` to make that clearer.

Reply:
Addressed in https://github.com/cypress-io/cypress/pull/19323/commits/7735c61e6fd46b84e371c8885fadebaa060b454b